### PR TITLE
Feature/#8 citation annotation

### PR DESCRIPTION
**Pull request by felixreimann**
fixes issue #8

### CODE REVIEW

File: opt4j-core/src/main/java/org/opt4j/core/config/annotations/Citation.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+import org.opt4j.core.start.Opt4JModule;
+
+/**
+ * The {@link Citation} can be used in {@link Opt4JModule}s to add citations to scientific works.
+ * 
+ * Example:
+ * 
+ * <pre>
+ * &#64;Citation(title     = "Opt4J: A Modular Framework for Meta-heuristic Optimization",
+ *               authors   = "Martin Lukasiewycz, Michael Glass, Felix Reimann, and J�rgen Teich",
```
Comment (michaelhglass) [inline opt4j-core/src/main/java/org/opt4j/core/config/annotations/Citation.java]: ss -> &szlig;
;)

File: opt4j-benchmarks/src/main/java/org/opt4j/benchmarks/dtlz/DTLZModule.java [hunk 2]
```diff
@@ -40,14 +37,14 @@
...
- * optimization test problems, Kalyanmoy Deb,Lothar Thiele, Marco Laumanns,
- * Eckart Zitzler, Proc. Congress Evolutionary Computation Volume 1 (May 2002)".
+ * Module for the DTLZ benchmarks as proposed in "Scalable multi-objective optimization test problems, Kalyanmoy Deb,
+ * Lothar Thiele, Marco Laumanns, Eckart Zitzler, Proc. Congress Evolutionary Computation Volume 1 (May 2002)".
  * 
  * @author lukasiewycz
  * 
  */
 @Info("DTLZ Problem Suite.  (n = m + k - 1)")
+@Citation(title = "Scalable multi-objective optimization test problems", authors = "Kalyanmoy Deb, Lothar Thiele, Marco Laumanns, and Eckart Zitzler", journal = "Proceedings of the 2002 Congress on Evolutionary Computation", volume = 1, month = MAY, year = 2002, doi = "10.1109/CEC.2002.1007032")
```
Comment (michaelhglass) [inline opt4j-benchmarks/src/main/java/org/opt4j/benchmarks/dtlz/DTLZModule.java]: Check given names

File: opt4j-benchmarks/src/main/java/org/opt4j/benchmarks/dtlz/DTLZModule.java [hunk 3]
```diff
@@ -40,14 +37,14 @@
 import org.opt4j.core.start.Constant;
 
 /**
- * Module for the DTLZ benchmarks as proposed in "Scalable multi-objective
- * optimization test problems, Kalyanmoy Deb,Lothar Thiele, Marco Laumanns,
- * Eckart Zitzler, Proc. Congress Evolutionary Computation Volume 1 (May 2002)".
+ * Module for the DTLZ benchmarks as proposed in "Scalable multi-objective optimization test problems, Kalyanmoy Deb,
```
Comment (michaelhglass) [inline opt4j-benchmarks/src/main/java/org/opt4j/benchmarks/dtlz/DTLZModule.java]: Check given names.

File: opt4j-benchmarks/src/main/java/org/opt4j/benchmarks/wfg/WFGModule.java [hunk 4]
```diff
@@ -40,14 +36,14 @@
 import org.opt4j.core.problem.ProblemModule;
 
 /**
- * Module for the WFG (Walking Fish Group) benchmarks, see "A Scalable
- * Multi-objective Test Problem Toolkit, Simon Huband, Luigi Barone, R. Lyndon
- * While, and Philip Hingston (EMO 2005)".
+ * Module for the WFG (Walking Fish Group) benchmarks, see "A Scalable Multi-objective Test Problem Toolkit, Simon
```
Comment (michaelhglass) [inline opt4j-benchmarks/src/main/java/org/opt4j/benchmarks/wfg/WFGModule.java]: Check given names.

File: opt4j-benchmarks/src/main/java/org/opt4j/benchmarks/wfg/WFGModule.java [hunk 5]
```diff
@@ -40,14 +36,14 @@
...
- * Multi-objective Test Problem Toolkit, Simon Huband, Luigi Barone, R. Lyndon
- * While, and Philip Hingston (EMO 2005)".
+ * Module for the WFG (Walking Fish Group) benchmarks, see "A Scalable Multi-objective Test Problem Toolkit, Simon
+ * Huband, Luigi Barone, R. Lyndon While, and Philip Hingston (EMO 2005)".
  * 
  * @author lukasiewycz
  * 
  */
 @Info("WFG Problem Suite. The number of search variables is n=k+l.")
+@Citation(title = "A Scalable Multi-objective Test Problem Toolkit", authors = "Simon Huband, Luigi Barone, R. Lyndon While, and Philip Hingston", journal = "Evolutionary Multi-Criterion Optimization", pageFirst = 280, pageLast = 295, year = 2005, month = PublicationMonth.UNKNOWN, doi = "10.1007/978-3-540-31880-4_20")
```
Comment (michaelhglass) [inline opt4j-benchmarks/src/main/java/org/opt4j/benchmarks/wfg/WFGModule.java]: Check given names.

File: opt4j-benchmarks/src/main/java/org/opt4j/benchmarks/zdt/ZDTModule.java [hunk 6]
```diff
@@ -39,15 +35,15 @@
 import org.opt4j.core.problem.ProblemModule;
 
 /**
- * The {@link ZDTModule} configures the ZDT benchmarks, see "Comparison of
- * Multiobjective Evolutionary Algorithms: Empirical Results, Eckart Zitzler,
- * Kalyanmoy Deb, Lothar Thiele, Evolutionary Computation Volume 8 , Issue 2
- * (June 2000)".
+ * The {@link ZDTModule} configures the ZDT benchmarks, see "Comparison of Multiobjective Evolutionary Algorithms:
```
Comment (michaelhglass) [inline opt4j-benchmarks/src/main/java/org/opt4j/benchmarks/zdt/ZDTModule.java]: Check given names.

File: opt4j-benchmarks/src/main/java/org/opt4j/benchmarks/zdt/ZDTModule.java [hunk 7]
```diff
@@ -39,15 +35,15 @@
...
- * (June 2000)".
+ * The {@link ZDTModule} configures the ZDT benchmarks, see "Comparison of Multiobjective Evolutionary Algorithms:
+ * Empirical Results, Eckart Zitzler, Kalyanmoy Deb, Lothar Thiele, Evolutionary Computation Volume 8 , Issue 2 (June
+ * 2000)".
  * 
  * @author lukasiewycz
  * 
  */
 @Info("ZDT Problem Suite.")
+@Citation(title = "Comparison of Multiobjective Evolutionary Algorithms: Empirical Results", authors = "Eckart Zitzler, Kalyanmoy Deb, Lothar Thiele", journal = "Evolutionary Computation Volume", volume = 8, number = 2, month = PublicationMonth.JUNE, year = 2000)
```
Comment (michaelhglass) [inline opt4j-benchmarks/src/main/java/org/opt4j/benchmarks/zdt/ZDTModule.java]: Check given names.

File: opt4j-optimizers/src/main/java/org/opt4j/optimizers/ea/Spea2Module.java [hunk 8]
```diff
@@ -33,6 +30,7 @@
  * 
  */
 @Info("The Strength Pareto Evolutionary Algorithm (SPEA II).")
+@Citation(title = "SPEA2: Improving the Strength Pareto Evolutionary Algorithm For Multiobjective Optimization", authors = "Eckart Zitzler, Marco Laumanns, and Lothar Thiele", journal = "Evolutionary Methods for Design, Optimisation, and Control", pageFirst = 19, pageLast = 26, year = 2002, month = UNKNOWN)
```
Comment (michaelhglass) [inline opt4j-optimizers/src/main/java/org/opt4j/optimizers/ea/Spea2Module.java]: Check given names.

File: opt4j-benchmarks/src/main/java/org/opt4j/benchmarks/dtlz/DTLZModule.java (hunk 2 above)
Comment (felixreimann) [reply opt4j-benchmarks/src/main/java/org/opt4j/benchmarks/dtlz/DTLZModule.java]: I did and found no problem. What is wrong?

Comment (michaelhglass) [reply opt4j-benchmarks/src/main/java/org/opt4j/benchmarks/dtlz/DTLZModule.java]: See overall comment: I suggest to stick to one naming convention. It seems the majority preferred J. Doe over Jane Doe so I marked all occurrences of the latter.

**Comment by felixreimann** [comment]
All names are now in the form "givenName surname, givenName surname, and givenName surname".

**Pull request merged**